llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py

```diff
@@ -177,21 +177,57 @@ def _get_all_kwargs(self, **kwargs) -> Dict[str, Any]:
                 all_kwargs.pop(key, None)
         return all_kwargs
 
-    @llm_chat_callback()
-    def chat(self, messages: Sequence[ChatMessage], **kwargs: Any) -> ChatResponse:
-        all_kwargs = self._get_all_kwargs(**{**self.additional_kwargs, **kwargs})
+    def _prepare_messages_before_chat(
```

Comment:
Can you move it to `utils.py` ? You can check [`anthropic`](https://github.com/run-llama/llama_index/blob/00e118c1c1cd5d587c45790b7ed515533279d2b1/llama-index-integrations/llms/llama-index-llms-anthropic/llama_index/llms/anthropic/utils.py#L54) or any other LLM integration for reference.

Reply:
Sure, this makes sense, done